polar-core/src/partial/partial.rs

```diff
@@ -864,27 +855,30 @@ mod test {
                k(y) if x = y and y == z and z = 1 and x = 2;"#,
         )?;
         let mut q = p.new_query_from_term(term!(call!("f", [sym!("x")])), false);
-        assert_partial_expression!(next_binding(&mut q)?, "x", "_this > 1 and _this = 1");
+        assert_partial_expression!(next_binding(&mut q)?, "x", "1 > 1");
         assert_query_done!(q);
 
         let mut q = p.new_query_from_term(term!(call!("g", [sym!("x")])), false);
-        assert_partial_expression!(next_binding(&mut q)?, "x", "_this > 1 and _this == 1");
+        assert_partial_expression!(next_binding(&mut q)?, "x", "1 > 1");
         assert_query_done!(q);
 
         let mut q = p.new_query_from_term(term!(call!("h", [sym!("x")])), false);
-        assert_partial_expression!(next_binding(&mut q)?, "x", "_this > 1 and _this = 2");
+        assert_partial_expression!(next_binding(&mut q)?, "x", "2 > 1");
         assert_query_done!(q);
 
         let mut q = p.new_query_from_term(term!(call!("i", [sym!("x")])), false);
-        assert_partial_expression!(next_binding(&mut q)?, "x", "_this > 1 and _this == 2");
+        assert_partial_expression!(next_binding(&mut q)?, "x", "2 > 1");
         assert_query_done!(q);
 
         let mut q = p.new_query_from_term(term!(call!("j", [sym!("y")])), false);
-        assert_partial_expression!(next_binding(&mut q)?, "y", "_this = 1 and _this == 1");
+        // This seems wrong, right? we have `x = 1` so we should just get `1`.
```

Comment:
Add note to grounding bug to make sure we look at this.

Reply:
is this done?

Reply:
yes